thread/thread.go

```diff
@@ -224,11 +218,35 @@ func tryToDeleteFile(filename string) {
 	}
 }
 
-// deleteOldestThreadFile deletes the single oldest file held by this thread.
+// pruneOldestThreadFiles deletes enough of the oldest files held by this
+// thread to free up bytes >= the size of the newest file.
+// It should only exceed the newest size by no more than the size of the last
+// deleted file.
 // It should only be called if the thread has at least one file (should be
 // checked by the caller beforehand).
-func (t *Thread) deleteOldestThreadFiles(n int) {
+func (t *Thread) pruneOldestThreadFiles() {
 	files := t.getSortedFiles()
+	firstName := files[len(files)-1]
+	firstSize := t.files[firstName].Size()
+	var delSize int64 = 0
+	delCnt := 0
+	for delSize < firstSize {
```

Comment:
I think you want `<=` here, otherwise if there's only one file in the `files` list could be problematic.